src/tools/vscode-language-client/src/workspace.ts

```diff
@@ -0,0 +1,199 @@
+/**
+ * @license
+ * Copyright 2019 Google LLC.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * Code distributed by Google as part of this project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+import * as path from 'path';
+import {
+  Disposable,
+  ExtensionContext,
+  languages,
+  OutputChannel,
+  TextDocument,
+  Uri,
+  window,
+  workspace,
+  WorkspaceConfiguration,
+  TextDocumentContentChangeEvent,
+  TextDocumentChangeEvent,
+  Diagnostic,
+  DiagnosticCollection
+} from 'vscode';
+import {
+  LanguageClient,
+  LanguageClientOptions,
+  RevealOutputChannelOn,
+  Executable,
+} from 'vscode-languageclient';
+
+const clientName = 'Arcs Language Server';
+
+export class ClientWorkspace {
+  private configuration: WorkspaceConfiguration;
+  // TODO: private readonly folder: WorkspaceFolder;
+  private disposables: Disposable[];
+  private lc: LanguageClient | null = null;
+  public outputChannel: OutputChannel;
+  private diagnostics: DiagnosticCollection;
+
+  constructor(outputChannel: OutputChannel) {
+    this.disposables = [];
+    this.configuration = workspace.getConfiguration();
+    this.outputChannel = outputChannel;
+    this.diagnostics = languages.createDiagnosticCollection('arcs');
+  }
+
+  public async start(context: ExtensionContext): Promise<boolean> {
+    if (!this.lspPath) {
+      // Warn the user
+      return false;
+    }
+    // Run the node server
+    const serverOptions: Executable = {
+      command: 'node',
+      args: [
+        this.lspPath,
+        '--stdio'
+        // '--log=~/.arcs-lsp-log-vscode'
```

Comment:
Delete?

Reply:
Ack. Wanted to get the logging working but will leave it for later.